Mathlib/LinearAlgebra/Multilinear/Basic.lean

```diff
@@ -252,6 +254,21 @@ def toLinearMap [DecidableEq ι] (m : ∀ i, M₁ i) (i : ι) : M₁ i →ₗ[R]
 #align multilinear_map.to_linear_map MultilinearMap.toLinearMap
 #align multilinear_map.to_linear_map_to_add_hom_apply MultilinearMap.toLinearMap_apply
 
+/-- This introduces the "derivative" of a multilinear map, as a linear map from
+`(i : ι) → M₁` to `M₂`. For continuous multilinear maps, this will indeed be the
+derivative.-/
+noncomputable def linearDeriv [DecidableEq ι] [Fintype ι] (f : MultilinearMap R M₁ M₂)
+(x : (i : ι) → M₁ i) : ((i : ι) → M₁ i) →ₗ[R] M₂ :=
+Finset.sum Finset.univ (fun (i : ι) => (f.toLinearMap x i).comp (LinearMap.proj i))
+
+@[simp]
+lemma linearDeriv_apply [DecidableEq ι] [Fintype ι] (f : MultilinearMap R M₁ M₂)
+    (x y : (i : ι) → M₁ i) : f.linearDeriv x y
+    = Finset.sum Finset.univ (fun (i : ι) => f (Function.update x i (y i))) := by
```

Comment:
It's usually best to put both sides of an `=` on the same line if you can, for readability:
```suggestion
    (x y : (i : ι) → M₁ i) :
    f.linearDeriv x y = ∑ i : ι, f (Function.update x i (y i)) := by
```
(and notation again!)